src/main/webapp/js/components/interface/drawer/TabbedDrawer.js

```diff
@@ -79,90 +77,125 @@ define(function (require) {
                 );
             }, this);
             return renderedLabels;
-        }
+        };
 
         // function to render all the childs, wrap each one of them in a div and manage with display
         // which child has to be visible.
         renderMyChildren() {
-            var renderedComponents = this.props.children.map(function(child, index) {
-                if(this.state.drawerOpened == true) {
-                    var ElementToRender = child;
-                    var MyElement = (this.state.selectedTab != index) ? <div className="hiddenComponent" key={index}><ElementToRender /></div> : <div key={index}><ElementToRender /></div>;
-
-                } else {
-                    var ElementToRender = child;
-                    var MyElement = <div key={index}><ElementToRender className="hiddenComponent" /></div>;
-                }
-                return (MyElement);
-            }, this);
-            return renderedComponents;
-        }
+            var paddingChildren = 45;
+            // Check for children since react will complain if we try to iterate an undefined object.
+            if (this.props.children != undefined) {
+                var renderedComponents = this.props.children.map(function (child, index) {
+                    var DrawerChild = undefined;
+                    var ComponentToRender = undefined;
+                    // I am relying on the type to distinguish between componentFactory usage in the
+                    // ComponentInitialization and the reacty way to use the TabbedDrawer.
+                    // If type is defined we are in the reacty way to use the component, for that reason I
+                    // need to take the type to redefine the component, since what I have as child is an object.
+                    if(child.type != undefined) {
+                        var DrawerChild = child.type;
+                        var properties = child.props;
+                    // If child.type is undefined we are in the scope of the componentFactory, this already
+                    // provides us the type directly, so we can reassign this to DrawerChild and declare
+                    // the child component we need into the tabbedDrawer.
+                    } else {
+                        var DrawerChild = child;
+                        // properties passed through the componentFactory method to declare the component
+                        // we extract the properties for the specific child component based on the order
+                        // these have been declared.
+                        var properties = this.props.childrenProps[index];
```

Comment:
@ddelpiano this `childrenProps` is what I was referring to. As I understand this is only needed if someone wants to use the tabbed drawer from the component factory, so it essentially keeps adding support for the old way of doing things. I'd be inclined to remove this and simplify the code, if someone wants to have nested components they really need to go full React. Thoughts? cc @adrianq

Reply:
I agree 100% just that I don't know the impact of this on the other extensions. Is the component factory deprecated? or is it used for certain components?

Reply:
No it's not deprecated (yet) but I don't want to expand its scope to go from adding single components to adding nested ones. VFB is migrating away from it, OSB is the only one lagging behind in this respect. But this change is only needed to propagate properties from parent to children which currently is only required when embedding the Python Console inside the tabbed drawer. Since NetPyNE is the only application using the Python Console there's nothing to worry about for the other extensions.

Reply:
@tarelli just pushed what discussed earlier, renderMyChildren handles the 2 cases separately, reacty way it clones the child one by one to inject the prop(s) required and defined into the TabbedDrawer wether the second case (ComponentFactory) instantiate the component required.
ChildrenProps removed as discussed since is not required.